core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,242 @@
+                  $scope.objectType));
+              $modalInstance.close({
+                solution: SolutionObjectFactory.createNew(
+                    $scope.tmpSolution.answerIsExclusive,
+                    answer,
+                    $scope.tmpSolution.explanation)
+              });
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function(result) {
+        var correctAnswer = result.solution.correctAnswer;
+        var currentStateName = editorContextService.getActiveStateName();
+        var state = explorationStatesService.getState(currentStateName);
+        try {
+          StateSolutionHelperService.verifyAndSaveAnswer(
+            explorationContextService.getExplorationId(),
+            state,
+            correctAnswer);
+        } catch (e) {
+          alertsService.addInfoMessage('That solution was invalid!');
+          $scope.openAddSolutionModal();
+        }
+        stateSolutionService.displayed = result.solution;
+      });
+    };
+
+    $scope.deleteSolution = function(evt) {
+      evt.stopPropagation();
+
+      alertsService.clearWarnings();
+      $modal.open({
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/exploration_editor/editor_tab/deleteSolutionModal.html'),
+        backdrop: true,
+        controller: [
+          '$scope', '$modalInstance',
+          function($scope, $modalInstance) {
+            $scope.reallyDelete = function() {
+              $modalInstance.close();
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function() {
+        stateSolutionService.displayed = null;
+        stateSolutionService.saveDisplayedValue();
+        StateSolutionHelperService.unsetSolutionIsValidFlag();
+      });
+    };
+
+    $scope.onComponentSave = function() {
+      alertsService.clearWarnings();
+      var explorationId = explorationContextService.getExplorationId();
+      var currentStateName = editorContextService.getActiveStateName();
+      var state = explorationStatesService.getState(currentStateName);
+      var interactionId = stateInteractionIdService.savedMemento;
+      var rulesServiceName =
+        angularNameService.getNameOfInteractionRulesService(
+          interactionId);
+      // Inject RulesService dynamically.
+      var rulesService = $injector.get(rulesServiceName);
+      var answer = stateSolutionService.savedMemento.correctAnswer;
+      try {
+        AnswerClassificationService.getMatchingClassificationResult(
+          explorationId, state, answer, true, rulesService).then(
+            function(result) {
+              if (editorContextService.getActiveStateName() !== (
```

Comment:
I keep seeing this logic repeated in multiple places. Can you reuse one of the verify() methods?